log/slow/parser.go

```diff
@@ -358,7 +358,11 @@ func (p *SlowLogParser) sendEvent(inHeader bool, inQuery bool) {
 		// Started parsing in header after Query_time.  Throw away event.
 		return
 	}
-
+	// If the query was logged because it took more than RateMaxQueryTime
+	// the rate should be ignored to prevent a wrong sampling
+	if p.opt.RateMaxQueryTime > 0 && p.event.TimeMetrics["Query_time"] > p.opt.RateMaxQueryTime {
+		p.event.RateLimit = 0
```

Comment:
As per Peter explanation:
We must not skip long queries, however we just need to make sure not to multiply them 
Ie one query took 5 minutes   with sampling 1/1000 we shuld not count it as 1000 of 5min queries as it is going to crew up results.